packages/fields/src/fields/slug/slug-view.tsx

```diff
@@ -7,9 +7,10 @@ import { useEffect, useRef } from '@wordpress/element';
  * Internal dependencies
  */
 import type { BasePost } from '../../types';
+import { getSlug } from './utils';
 
 const SlugView = ( { item }: { item: BasePost } ) => {
-	const slug = item.slug;
+	const slug = item ? getSlug( item ) : '';
```

Comment:
`getSlug` will always return something, even if it's the `item.id`. Do we need to cover against not returning anything (empty string)?

Reply:
I didn't have time to investigate why, but during the loading of the page, `item` is a boolean. You can add print `data` here: https://github.com/WordPress/gutenberg/blob/d871796b8fb8b399f240cb8835ff47a31e8f50ea/packages/dataviews/src/dataforms-layouts/panel/index.tsx#L59-L68




https://github.com/user-attachments/assets/00b24946-2068-476b-943e-c3371b5d2553

Reply:
I see. Can you instead check if item is an object? It just happens to be a boolean, but that check would pass if it's a string only to break later in `getSlug`.

A follow-up is that we need a better management of loading states for `DataForm`, like we have for `DataViews`. The issue is that the data is not yet loaded. For `DataViews` we have `isLoading` prop, it sounds like we need the same for `DataForm`. Is this something you'd be able to follow-up on?

Reply:
Actually, wouldn't this logic be best placed in `getValue` instead of the `getSlug` utility? That way edit & view components would have access to it.

Reply:
Improved with 51e08fe4754432017a7dd25c6c9d7f0031d67ac8.

> Is this something you'd be able to follow-up on?

Sure! I created an issue https://github.com/WordPress/gutenberg/issues/66599.

Reply:
> Actually, wouldn't this logic be best placed in getValue instead of the getSlug utility? That way edit & view components would have access to it.

What do you think of this?

Reply:
It is a good idea, but currently we are not passing to the render component, the `field` prop. I would create a dedicated issue and craft a dedicated PR for this. If you think that it is not necessary, I'm happy to implement it in this PR too.

Reply:
Oh, you're right. I thought we did at some point? 🤔 Let's ship and move on.

Reply:
I just [looked this up](https://github.com/WordPress/gutenberg/blob/trunk/packages/dataviews/src/types.ts#L182) and it's confusing why the Edit function receives data (Item), field, onChange, _hideLabelFromVision_ but the View function only receives the item (Item). There's also different names (data vs item) for the same prop across functions. I think we need to consolidate and:

1. The data prop should be called `item` in both places.
2. The `field` should be passed down as a prop in both places.

Reply:
I created a dedicated issue: https://github.com/WordPress/gutenberg/issues/66616

Reply:
related https://github.com/WordPress/gutenberg/pull/67577/